iothub/device/tests/DeviceClientTests.cs

```diff
@@ -1773,9 +1775,291 @@ public void DeviceClient_InitWithHttpTransportButNoModelId_DoesNotThrow()
                 .NotThrow();
         }
 
+
+        [Description("Testing current behavior, The logically correct expected exception should be OperationCancelledException but current expected exception is IotHubCommunicationException since we try to avoid behavior change.")]
+        [TestMethod]
+        [ExpectedException(typeof(IotHubCommunicationException))]
+        public async Task DeviceClient_ReceiveAsync_Cancelled_ThrowsCorrectException()
```

Comment:
What do you think about the method name illustrating that we are verifying and preserving legacy behavior?

Reply:
I like it ! 
Decided on `DeviceClient_ReceiveAsync_MaintainLegacyExecptionBehavior`